app/src/main/java/org/grakovne/lissen/AutoBookPlayerService.kt

```diff
@@ -0,0 +1,148 @@
+
+package org.grakovne.lissen
+
+import android.app.PendingIntent
+import android.content.Intent
+import android.net.Uri
+
+import androidx.annotation.OptIn
+import androidx.media3.common.MediaItem
+import androidx.media3.common.MediaMetadata
+import androidx.media3.common.MimeTypes
+import androidx.media3.common.util.UnstableApi
+import androidx.media3.exoplayer.ExoPlayer
+import androidx.media3.session.LibraryResult
+import dagger.hilt.android.AndroidEntryPoint
+
+import org.grakovne.lissen.content.LissenMediaProvider
+
+import org.grakovne.lissen.widget.MediaRepository
+
+import javax.inject.Inject
+
+import androidx.media3.session.MediaLibraryService
+import androidx.media3.session.MediaSession
+import com.google.common.collect.ImmutableList
+import com.google.common.util.concurrent.Futures
+import com.google.common.util.concurrent.ListenableFuture
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+@AndroidEntryPoint
+class AutoBookPlayerService : MediaLibraryService() {
+
+    private lateinit var mediaSession: MediaLibrarySession
+    private lateinit var books: MutableList<MediaItem>
+
+    @Inject
+    lateinit var exoPlayer: ExoPlayer
+
+    @Inject
+    lateinit var mediaChannel: LissenMediaProvider
+
+    @Inject
+    lateinit var mediaRepository: MediaRepository
+
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        CoroutineScope(Dispatchers.IO).launch{
+            books = getBooksMediaItems()
+        }
+        return super.onStartCommand(intent, flags, startId)
+
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+
+        mediaSession = MediaLibrarySession.Builder(
+                this,
+                exoPlayer,
+                MediaLibrarySessionCallback()
+        ).build()
+    }
+
+    override fun onGetSession(controllerInfo: MediaSession.ControllerInfo): MediaLibrarySession? {
+        return mediaSession
+    }
+
```

Comment:
I found here (https://developer.android.com/reference/androidx/media3/session/MediaLibraryService.MediaLibrarySession) that `MediaLibrarySession` is an extension of `MediaSession` and I see that it has an extra attribute `MediaLibrarySession.Callback` that I imagine (as documentation is pretty much non-existent for Media3 Android Auto) will allow us to control the music from Android Auto. 

Obviously it complains:
```
Caused by: java.lang.IllegalStateException: Session ID must be unique. ID=
                                                                                                    	at androidx.media3.session.MediaSession.<init>(MediaSession.java:732)
                                                                                                    	at androidx.media3.session.MediaLibraryService$MediaLibrarySession.<init>(MediaLibraryService.java:725)
                                                                                                    	at androidx.media3.session.MediaLibraryService$MediaLibrarySession$Builder.build(MediaLibraryService.java:703)
                                                                                                    	at org.grakovne.lissen.AutoBookPlayerService.onCreate(AutoBookPlayerService.kt:62)
```

The only example I have found is this:
https://github.com/androidx/media/issues/204#issuecomment-1855905050